src/app/clusters/descriptor/descriptor.cpp

```diff
@@ -165,16 +159,15 @@ CHIP_ERROR DescriptorAttrAccess::ReadDeviceAttribute(EndpointId endpoint, Attrib
 {
     CHIP_ERROR err = aEncoder.EncodeList([&endpoint](const auto & encoder) -> CHIP_ERROR {
         Descriptor::Structs::DeviceTypeStruct::Type deviceStruct;
-        CHIP_ERROR err2;
 
-        auto deviceTypeList = emberAfDeviceTypeListFromEndpoint(endpoint, err2);
-        ReturnErrorOnFailure(err2);
+        auto deviceType = InteractionModelEngine::GetInstance()->GetDataModelProvider()->FirstDeviceType(endpoint);
 
-        for (auto & deviceType : deviceTypeList)
+        while (deviceType.has_value())
         {
-            deviceStruct.deviceType = deviceType.deviceId;
-            deviceStruct.revision   = deviceType.deviceVersion;
+            deviceStruct.deviceType = deviceType->deviceTypeId;
+            deviceStruct.revision   = deviceType->deviceTypeVersion;
```

Comment:
Pre-existing, but we should rename deviceTypeVersion to deviceTypeRevision to match the spec.  followup for that, I guess.

Reply:
Changed it for this struct, but looks like we are using Version in ember codes.